recipes/imagemgagick/all/conanfile.py

```diff
@@ -0,0 +1,349 @@
+               "openexr": [True, False],
+               "jpeg": [True, False],
+               "openjp2": [True, False],
+               "png": [True, False],
+               "tiff": [True, False],
+               "webp": [True, False],
+               "xml": [True, False],
+               "freetype": [True, False],
+               "utilities": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "hdri": True,
+                       "quantum_depth": 16,
+                       "zlib": True,
+                       "bzlib": True,
+                       "lzma": False,
+                       "lcms": False,
+                       "openexr": False,
+                       "jpeg": True,
+                       "openjp2": False,
+                       "png": True,
+                       "tiff": True,
+                       "webp": False,
+                       "xml": False,
+                       "freetype": True,
+                       "utilities": True}
+
+    _source_subfolder = "ImageMagick"  # name is important, VisualMagick uses relative paths to it
+    _build_subfolder = "build_subfolder"
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os == 'Windows' and self.settings.compiler == 'gcc' and os.name == 'nt'
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == 'Visual Studio'
+
+    @property
+    def _modules(self):
+        return ['Magick++', 'MagickWand', 'MagickCore']
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires('pango/1.48.5')
+        if self.options.zlib:
+            self.requires('zlib/1.2.11')
+        if self.options.bzlib:
+            self.requires('bzip2/1.0.8')
+        if self.options.lzma:
+            self.requires('xz_utils/5.2.5')
+        if self.options.lcms:
+            self.requires('lcms/2.11')
+        if self.options.openexr:
+            self.requires('openexr/2.5.5')
+        if self.options.jpeg:
+            self.requires('libjpeg/9d')
+        if self.options.openjp2:
+            self.requires('openjpeg/2.4.0')
+        if self.options.png:
+            self.requires('libpng/1.6.37')
+        if self.options.tiff:
+            self.requires('libtiff/4.2.0')
+        if self.options.webp:
+            self.requires('libwebp/1.2.0')
+        if self.options.xml:
+            self.requires('libxml2/2.9.10')
+        if self.options.freetype:
+            self.requires('freetype/2.10.4')
+
+    def source(self):
+        tools.get("{0}/archive/{1}.tar.gz".format(self.homepage, self.version))
+        extracted_dir = "ImageMagick-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+        tools.get('https://github.com/ImageMagick/VisualMagick/archive/refs/heads/main.zip')
+        os.rename('VisualMagick-main', 'VisualMagick')
```

Comment:
source url are listed in conandata.yml, please have a look to others recipes in CCI.
Never rely on head, it's not reproducible.